webapp/src/modules/asset/sagas.js

```diff
@@ -18,38 +24,55 @@ import { fetchAssetAcceptedBidsRequest } from 'modules/bid/actions'
 import { getData as getEstates } from 'modules/estates/selectors'
 
 export function* assetSaga() {
-  yield takeEvery(FETCH_ASSET, handleFetchAsset)
+  yield takeEvery(FETCH_ASSET_REQUEST, handleFetchAsset)
+  yield takeEvery(NAVIGATE_TO_ASSET, handleNavigateToAsset)
   yield takeEvery(FETCH_ASSET_LISTING_HISTORY, handleFetchAssetListingHistory)
 }
 
 function* handleFetchAsset(action) {
+  const { assetId, assetType } = action
+
+  try {
+    const asset = yield fetchAsset(assetId, assetType)
+    yield put(fetchAssetSuccess(asset, assetType))
+  } catch (error) {
+    yield put(fetchAssetFailure(assetId, assetType, error.message))
+  }
+}
+
+function* handleNavigateToAsset(action) {
+  const { assetId, assetType } = action
+
+  yield fetchAsset(assetId, assetType) // fill state
+  yield put(push(locations.assetDetail(assetId, assetType)))
+}
+
+function* handleFetchAssetListingHistory(action) {
   const { asset, assetType } = action
+  // TODO: fetch publications (?) and mortgages
+  yield put(fetchAssetAcceptedBidsRequest(asset, assetType))
+}
 
+export function* fetchAsset(assetId, assetType) {
+  // TODO: Check for _FAILURE cases
```

Comment:
It is a pending or something for the future that doesn't need attention now?

Reply:
Yes this code was already like this (I just moved it). What I've ment by that code is to check what happens when instead of firing `FETCH_PARCEL_SUCCESS or FETCH_ESTATE_SUCCESS` we fire the `_FAILURE` versions, for example if the server goes down.

I think it merits it's own PR